src/style/colors.ts

```diff
@@ -220,38 +220,19 @@ export class Colors {
     return this.shouldSupportDarkMode && Scheme.getSchemeType() === 'dark' ? _.reverse(palette) : palette;
   });
 
-  generateDesignTokens(color: string, mode: 'light' | 'dark' = 'light') {
-    return mode === 'light' ? this.generateLightModeTokens(color) : this.generateDarkModeTokens(color);
-  }
-
-  private generateLightModeTokens(color: string) {
-    const colorPalette: string[] = this.generatePalette(color);
-    const color30 = colorPalette[2];
-    const color50 = colorPalette[4];
-    const color70 = colorPalette[6];
-    const color80 = colorPalette[7];
-
-    const mainColor = this.isDark(color) ? color : color30;
-
-    return {
-      $backgroundPrimaryHeavy: mainColor,
-      $backgroundPrimaryLight: color80,
-      $backgroundPrimaryMedium: color70,
-      $iconPrimary: mainColor,
-      $iconPrimaryLight: color50,
-      $textPrimary: mainColor,
-      $outlinePrimary: mainColor
-    };
-  }
-
-  private generateDarkModeTokens(color: string) {
-    const colorPalette: string[] = _.reverse(this.generatePalette(color));
+  private generateDesignTokens({primaryColor, dark}: {primaryColor: string; dark?: boolean}) {
+    const colorPalette: string[] = dark
+      ? _.reverse(this.generatePalette(primaryColor))
```

Comment:
You can generate the palette outside of the condition and reverse the array if it is for a dark palette, this way you save the code duplication

Reply:
I actually tried it and found it less readable that `generateDesignTokens()` will need to get the `primaryColor`, the `palette`, and the `dark` boolean as parameters. it gets all the logic out of the method. 
And I think it's not that bad to call it twice since it happens only once when the app starts.

Reply:
I agree that the palette should be generated in the method, but you can still do it once and only reverse the array if dark = true. For example:
`let colorPalette = this.generatePalette(primaryColor);
  if (dark) {
    colorPalette = _.reverse(colorPalette);
  }
`

Reply:
Ohh right!
done